### PR TITLE
Bitemp vanila case source from/thru columns refactor

### PR DESCRIPTION
**Pull request by chloeminkyung**
#### What type of PR is this?
Refactor/Optimize

#### What does this PR do / why is it needed ?
- Refactor/optimize Bitemporal Delta ingest mode vanila case sql

origin SQL: INSERT INTO main (id, name, **validty_from_source, validity_through_source**, ..., **validity_from_target, validity_through_target**) (SELECT stage.id, stage.name, **stage.validity_from_reference, stage.validity_through_reference**, ..., **stage.validity_from_reference, stage.validity_through_reference**)

optimized SQL: INSERT INTO main (id, name, **validity_from_target, validity_through_target**, ...) (SELECT stage.id, stage.name, **stage.validity_from_reference, stage.validity_through_reference**, ...)

Removed repeated redundant columns

#### Does this PR introduce a user-facing change?
No

### CODE REVIEW

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/util/LogicalPlanUtils.java [hunk 1]
```diff
@@ -215,6 +215,20 @@ public static void removeField(List<Value> fieldsList, String fieldName)
         );
     }
 
+    public static void replaceField(List<Value> fieldsList, String oldFieldName, String newFieldName)
+    {
+        fieldsList.forEach(field ->
+        {
+            if (field instanceof FieldValue && ((FieldValue) field).fieldName().equalsIgnoreCase(oldFieldName))
```
Comment (kumuwu) [inline legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/util/LogicalPlanUtils.java:222]: Should we use equalsIgnoreCase or just equals?

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/util/LogicalPlanUtils.java [hunk 2]
```diff
@@ -215,6 +215,20 @@ public static void removeField(List<Value> fieldsList, String fieldName)
...
+            {
+                if (((FieldValue) field).fieldName().equalsIgnoreCase(oldFieldName))
```
Comment (kumuwu) [inline legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/util/LogicalPlanUtils.java:224]: This looks the same as the second part of the outer condition

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-ansi/src/test/java/org/finos/legend/engine/persistence/components/ingestmode/bitemporal/BitemporalDeltaWithBatchIdTest.java [hunk 3]
```diff
@@ -222,23 +218,23 @@ void testMilestoningSourceSpecifiesFromAndThroughWithUpperCase()
...
+                "\"BATCH_ID_OUT\") " +
+                "(SELECT stage.\"ID\",stage.\"NAME\",stage.\"AMOUNT\",stage.\"VALIDITY_FROM_REFERENCE\",stage.\"VALIDITY_THROUGH_REFERENCE\"," +
+                "stage.\"DIGEST\",(SELECT COALESCE(MAX(batch_metadata.\"TABLE_BATCH_ID\"),0)+1 FROM BATCH_METADATA as batch_metadata WHERE batch_metadata.\"TABLE_NAME\" = 'main')" +
+                ",999999999 " +
+                "FROM \"MYDB\".\"STAGING\" as stage WHERE NOT (EXISTS " +
+                "(SELECT * FROM \"MYDB\".\"MAIN\" as sink " +
+                "WHERE (sink.\"BATCH_ID_OUT\" = 999999999) " +
+                "AND (sink.\"DIGEST\" = stage.\"DIGEST\") " +
+                "AND ((sink.\"ID\" = stage.\"ID\") " +
+                "AND (sink.\"NAME\" = stage.\"NAME\")) " +
```
Comment (kumuwu) [inline legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-ansi/src/test/java/org/finos/legend/engine/persistence/components/ingestmode/bitemporal/BitemporalDeltaWithBatchIdTest.java:236]: Indentation accidentally changed

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/BitemporalDeltaPlanner.java [hunk 4]
```diff
@@ -135,7 +136,7 @@ class BitemporalDeltaPlanner extends BitemporalPlanner
             this.dataFields.removeIf(field -> field.fieldName().equals(ingestMode.digestField()));
             this.dataFields.removeIf(field -> field.fieldName().equals(sourceValidDatetimeFrom.fieldName()));
 
-            this.primaryKeys.removeIf(fieldName -> fieldName.equals(sourceValidDatetimeFrom.fieldName()));
+            this.primaryKeys.removeIf(fieldName -> fieldName.equals(targetValidDatetimeFrom.fieldName()));
```
[the remark this answers is not in the document]
Comment (chloeminkyung) [reply legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/BitemporalDeltaPlanner.java]: Oops - yes, changed it by mistake

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/util/LogicalPlanUtils.java (hunk 1 above)
Comment (chloeminkyung) [reply legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/util/LogicalPlanUtils.java]: fixed

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/util/LogicalPlanUtils.java [hunk 5]
```diff
@@ -215,6 +215,20 @@ public static void removeField(List<Value> fieldsList, String fieldName)
         );
     }
 
+    public static void replaceField(List<Value> fieldsList, String oldFieldName, String newFieldName)
+    {
+        fieldsList.forEach(field ->
+        {
+            if (field instanceof FieldValue && ((FieldValue) field).fieldName().equals(oldFieldName))
+            {
+                if (((FieldValue) field).fieldName().equals(oldFieldName))
```
Comment (prasar-ashutosh) [inline legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/util/LogicalPlanUtils.java:224]: Why double checking here ?

Comment (chloeminkyung) [reply legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/util/LogicalPlanUtils.java]: removed! thanks

**Review by kumuwu — APPROVED** [review]

File: legend-engine-xt-persistence-test-runner/src/test/resources/bitemporal-delta/batch_id_and_date_time_based/persistence_no_del_ind_user_specifies_from_and_thru.txt [hunk 6]
```diff
@@ -113,7 +113,7 @@ Persistence test::TestPersistence
...
+                   data: '[{"ID":1, "AMOUNT":100, "FROM_Z": "2022-01-01 00:00:00.0", "THRU_Z": "2022-01-02 00:00:00.0", "FROM_Z": "2022-01-01 00:00:00.0", "THRU_Z": "2022-01-02 00:00:00.0", "BATCH_IN":1, "BATCH_OUT":999999999},{"ID":2, "AMOUNT":200, "FROM_Z": "2022-01-01 00:00:00.0", "THRU_Z": "2022-01-02 00:00:00.0", "FROM_Z": "2022-01-01 00:00:00.0", "THRU_Z": "2022-01-02 00:00:00.0", "BATCH_IN":1, "BATCH_OUT":999999999},{"ID":3, "AMOUNT":400, "FROM_Z": "2022-01-01 00:00:00.0", "THRU_Z": "2022-01-02 00:00:00.0", "FROM_Z": "2022-01-01 00:00:00.0", "THRU_Z": "2022-01-02 00:00:00.0", "BATCH_IN":1, "BATCH_OUT":999999999}]';
```
Comment (kumuwu) [inline legend-engine-xt-persistence-test-runner/src/test/resources/bitemporal-delta/batch_id_and_date_time_based/persistence_no_del_ind_user_specifies_from_and_thru.txt:116]: Believe we can remove one set of "FROM_Z" and "THRU_Z" from each assert, for now looks like it's repeated twice for each row

Comment (chloeminkyung) [reply legend-engine-xt-persistence-test-runner/src/test/resources/bitemporal-delta/batch_id_and_date_time_based/persistence_no_del_ind_user_specifies_from_and_thru.txt]: fixed in the latest commit!

**Review by rafaelbey — APPROVED** [review]

**Pull request merged**